File: Sources/LegacyClient/Public/Sources/Native/Biome.ts
import type { ColorCode } from "../../../Public/Sources/Utils/Color";

export const enum Biome {
    GARDEN,
    DESERT,
    OCEAN,
}

type BiomeDisplayName = Capitalize<string>;

export const BIOME_DISPLAY_NAME = {
    [Biome.GARDEN]: "Garden",
    [Biome.DESERT]: "Desert",
    [Biome.OCEAN]: "Ocean",
} as const satisfies Record<Biome, BiomeDisplayName>;

export const BIOME_GAUGE_COLORS = {
    [Biome.GARDEN]: "#1ea761",
    [Biome.DESERT]: "#ecdcb8",
    [Biome.OCEAN]: "#4e77a7",
} as const satisfies Record<Biome, ColorCode>;

File: Sources/Client/private/src/Florr/Native/Biome.zig
const std = @import("std");
const meta = std.meta;
const Color = @import("../../WebAssembly/Interop/Canvas/Color.zig");

pub const Biome = enum(u8) {
    garden,
    desert,
    ocean,
};

const biome_names = std.StaticStringMap([]const u8).initComptime(.{
    .{ @tagName(Biome.garden), "Garden" },
    .{ @tagName(Biome.desert), "Desert" },
    .{ @tagName(Biome.ocean), "Ocean" },
});

const biome_colors = std.StaticStringMap(Color).initComptime(.{
    .{ @tagName(Biome.garden), Color.comptimeFromHexColorCode("#1ea761") },
    .{ @tagName(Biome.desert), Color.comptimeFromHexColorCode("#ecdcb8") },
    .{ @tagName(Biome.ocean), Color.comptimeFromHexColorCode("#4e77a7") },
});

pub inline fn getBiomeName(biome: Biome) ?[]const u8 {
    return biome_names.get(@tagName(biome));
}

pub inline fn getBiomeColor(biome: Biome) ?Color {
    return biome_colors.get(@tagName(biome));
}
